src/CovidSim.cpp

```diff
@@ -480,8 +361,45 @@ int main(int argc, char* argv[])
 	fprintf(stderr, "Model finished\n");
 }
 
+void parse_bmp_option(std::string const& input) {
+	// make copy and convert input to lowercase
+	std::string input_copy = input;
+	std::transform(input_copy.begin(), input_copy.end(), input_copy.begin(), [](unsigned char c){ return std::tolower(c); });
+
+	if (input_copy.compare("png") == 0) {
+#if defined(IMAGE_MAGICK) || defined(_WIN32)
+		P.BitmapFormat = BitmapFormats::PNG;
+#else
+		std::cerr << "PNG Bitmaps not supported - please build with Image Magic or WIN32 support" << std::endl;
+		std::exit(1);
+#endif
+	}
+	else if (input_copy.compare("bmp") == 0) {
+		P.BitmapFormat = BitmapFormats::BMP;
+	}
+	else {
+		std::cerr << "Unrecognised bitmap format: " << input_copy << std::endl;
+		std::exit(1);
+	}
+}
+
+void parse_intervention_file_option(std::string const& input) {
+	std::string output;
+	parse_read_file(input, output);
+	InterventionFiles.emplace_back(output);
+}
+
+void parse_save_snapshot_option(std::string const& input) {
+	auto sep = input.find_first_of(',');
+	if (sep == std::string::npos) {
+		std::cerr << "Argument value didn't use a ',': " << input << std::endl;
```

Comment:
This message could be improved - what should the argument look like?

Reply:
New comment should be better now.